andcolorpicker/src/main/java/codes/side/andcolorpicker/rgb/RGBColorPickerSeekBar.kt

```diff
@@ -0,0 +1,328 @@
+        return
+      }
+      _mode = value
+      refreshProperties()
+      refreshProgressFromCurrentColor()
+      refreshProgressDrawable()
+      refreshThumb()
+    }
+
+  private var coloringModeInitialized = false
+  private var _coloringMode: ColoringMode? = null
+  var coloringMode: ColoringMode
+    get() {
+      return requireNotNull(_coloringMode) { "Coloring mode is not initialized yet" }
+    }
+    set(value) {
+      coloringModeInitialized = true
+      if (_coloringMode == value) {
+        return
+      }
+      _coloringMode = value
+      refreshProgressDrawable()
+      refreshThumb()
+    }
+
+  init {
+    init(attrs)
+  }
+
+  private fun init(attrs: AttributeSet? = null) {
+    val typedArray = context.theme.obtainStyledAttributes(
+      attrs,
+      R.styleable.RGBColorPickerSeekBar,
+      0,
+      0
+    )
+
+    mode = Mode.values()[typedArray.getInteger(
+      R.styleable.RGBColorPickerSeekBar_rgbMode,
+      DEFAULT_MODE.ordinal
+    )]
+    coloringMode = ColoringMode.values()[typedArray.getInteger(
+      R.styleable.RGBColorPickerSeekBar_rgbColoringMode,
+      DEFAULT_COLORING_MODE.ordinal
+    )]
+
+    typedArray.recycle()
+  }
+
+  override fun setMin(min: Int) {
+    if (modeInitialized && min != mode.minProgress) {
+      throw IllegalArgumentException("Current mode supports ${mode.minProgress} min value only")
+    }
+    super.setMin(min)
+  }
+
+  override fun setMax(max: Int) {
+    if (modeInitialized && max != mode.maxProgress) {
+      throw IllegalArgumentException("Current mode supports ${mode.maxProgress} max value only")
+    }
+    super.setMax(max)
+  }
+
+  override fun updateInternalPickedColorFrom(value: IntegerRGBColor) {
+    super.updateInternalPickedColorFrom(value)
+    internalPickedColor.setFrom(value)
+  }
+
+  override fun refreshProperties() {
+    super.refreshProperties()
+    if (!modeInitialized) {
+      return
+    }
+    max = mode.maxProgress
+  }
+
+  override fun refreshProgressDrawable() {
+    super.refreshProgressDrawable()
+
+    if (!coloringModeInitialized || !modeInitialized) {
+      return
+    }
+
+    ((progressDrawable as LayerDrawable).getDrawable(0) as GradientDrawable).colors = when (mode) {
+      Mode.MODE_R -> {
+        when (coloringMode) {
+          ColoringMode.PURE_COLOR -> PURE_RED_COLOR_CHECKPOINTS
+          ColoringMode.OUTPUT_COLOR -> TODO()
+          ColoringMode.PLAIN_COLOR -> PLAIN_RED_COLOR_CHECKPOINTS
+        }
+      }
+      Mode.MODE_G -> {
+        when (coloringMode) {
+          ColoringMode.PURE_COLOR -> PURE_GREEN_COLOR_CHECKPOINTS
+          ColoringMode.OUTPUT_COLOR -> TODO()
+          ColoringMode.PLAIN_COLOR -> PLAIN_GREEN_COLOR_CHECKPOINTS
+        }
+      }
+      Mode.MODE_B -> {
+        when (coloringMode) {
+          ColoringMode.PURE_COLOR -> PURE_BLUE_COLOR_CHECKPOINTS
+          ColoringMode.OUTPUT_COLOR -> TODO()
+          ColoringMode.PLAIN_COLOR -> PLAIN_BLUE_COLOR_CHECKPOINTS
+        }
+      }
```

Comment:
It will be possible to read checkpoints from color mode on enum move.

Reply:
Resolved